### PR TITLE
[Prot Paladin] Add NoDamageShieldOfTheRighteous Analyzer

### PR DESCRIPTION
**Pull request by hpabst**
## Notes

- Add analyzer for tracking SotR casts which failed to hit any enemies.
- Resolves issue #3872 
- [Log used in testing](https://www.warcraftlogs.com/reports/ApK82yWNPF7bQBft#fight=39&source=27)

## Testing

- `yarn lint`, `yarn build`, `yarn test:integration`

### Statistic:
2020-10-15 19_09_07-Heroic N'Zoth the Corruptor - Kill (8_21) by Enst in anal fluids - WoWAnalyzer

### Suggestion:
2020-10-15 19_08_36-Heroic N'Zoth the Corruptor - Kill (8_21) by Enst in anal fluids - WoWAnalyzer

### CODE REVIEW

File: src/parser/paladin/protection/modules/features/Checklist/Module.js [hunk 1]
```diff
@@ -21,6 +22,7 @@ class Checklist extends BaseChecklist {
     abilities: Abilities,
     preparationRuleAnalyzer: PreparationRuleAnalyzer,
     shieldOfTheRighteous: ShieldOfTheRighteous,
+    noDamateSOTR: NoDamageShieldOfTheRighteous,
```
Comment (emallson) [inline src/parser/paladin/protection/modules/features/Checklist/Module.js]: typo

File: src/parser/paladin/protection/modules/features/Checklist/Module.js [hunk 2]
```diff
@@ -39,6 +41,7 @@ class Checklist extends BaseChecklist {
           ...this.preparationRuleAnalyzer.thresholds,
           consecration: this.consecration.uptimeSuggestionThresholds,
           shieldOfTheRighteous: this.shieldOfTheRighteous.suggestionThresholds,
+          noDamateSOTR: this.noDamateSOTR.hitRatioSuggestionThresholds,
```
Comment (emallson) [inline src/parser/paladin/protection/modules/features/Checklist/Module.js]: typo

**Review by emallson — CHANGES_REQUESTED** [review]
couple typos i'd like fixed but other than that it looks good :+1:

**Review by emallson — APPROVED** [review]

**Comment by emallson** [comment]
@hpabst looks like you missed one: https://github.com/WoWAnalyzer/WoWAnalyzer/runs/1266048194

**Comment by hpabst** [comment]
> @hpabst looks like you missed one: https://github.com/WoWAnalyzer/WoWAnalyzer/runs/1266048194

Yup, just fixed and rerunning the tests locally.

**Pull request merged**